pandas/tests/frame/test_subclass.py

```diff
@@ -210,3 +210,17 @@ def test_subclass_align_combinations(self):
         tm.assert_series_equal(res1, exp2)
         tm.assertIsInstance(res2, tm.SubclassedDataFrame)
         tm.assert_frame_equal(res2, exp1)
+
+    def test_subclass_sparse_slice(self):
+        ssdf = tm.SubclassedSparseDataFrame([[1, 2, 3], [4, 5, 6], [7, 8, 9]])
+        ssdf.testattr = "testattr"
+
+        tm.assertIsInstance(ssdf.loc[:2], tm.SubclassedSparseDataFrame)
```

Comment:
As far as I can tell, `tm.assert_sp_frame_equal` only asserts both are instances of `SparseDataFrame` and not that they are the same type. What I meant to fix and test is only preserving the types when slicing. Should I add (not replace with) an additional check with `tm.assert_sp_frame_equal` anyway?